lib/auth.js

```diff
@@ -48,6 +55,66 @@ Auth.prototype.bearer = function (bearer, sendImmediately) {
   }
 }
 
+
+Auth.prototype.negotiate = function(request, authHeader, sendImmediately) {
+  var self = this
+
+  if (kerberos === undefined) {
+    // Nothing to do if we don't have Kerberos
+    // or a current credential cache
+    return
+  }
+
+  self.hasAuth = true
+
+  if (!sendImmediately) {
```

Comment:
I installed your setup on a new Docker ubuntu image.  I think you've got most of the way there.  What did you use as the default realm name, I just picked to use `EXAMPLE.COM`

Added an entry for `kdc.example.com` to point to localhost in /etc/hosts.

Then ran:

 ```
# I'm running as root in the container, so I don't have sudo before every command.
# This will set a password for the realm.
echo "test123" | krb5_newrealm

# Add a root principal, again with a silly password
kadmin.local -q "addprinc -pw test123 root@EXAMPLE.COM"

# Authenticate into the realm, this will establish a TGT
echo "test123" | kinit root@EXAMPLE.COM

# Create the service principal for the HTTP service, localhost is okay after all.  It will implicitly have the default realm name appended on to it.  So the real principal will be HTTP/localhost@EXAMPLE.COM
kadmin.local -q "addprinc -randkey HTTP/localhost"

# Add the service principal to root's keytab.
kadmin.local -q "ktadd HTTP/localhost"

# Then run the tests and everything passes.
npm run test
```

So yes the tests work for me too, great job! :tada: